python/pyarrow/tests/test_table.py

```diff
@@ -1009,6 +1009,21 @@ def test_table_from_pydict():
         pa.Table.from_pydict({'c0': [0, 1, 2]},
                              schema=pa.schema([("c0", pa.string())]))
 
+    # Missing schema fields from the passed mapping
+    with pytest.raises(KeyError, match="doesn\'t contain.* c\, d"):
```

Comment:
This is already tested below (test_table_from_pydict_schema), so maybe update the `match` check there instead

Reply:
It tests multiple missing keys, besides below is a parametrized test case and this is just a sanity check.

Reply:
You can also update the other test to have 2 missing fields, or remove the other test (and move the remaining part about less fields in the schema than in the data here), if you prefer. I personally just think there is no need tin testing the exact same thing twice (our tests are already difficult enough to navigate and find what is tested where).

This is all a minor comment, though, so see what you do with it ;)